engine/src/test/java/org/hibernate/search/test/dsl/SortDSLTest.java

```diff
@@ -0,0 +1,809 @@
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test
+	public void singleField_missingValue_use() throws Exception {
+		Query query = builder().all().createQuery();
+
+		Sort sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 1, 2, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.asc()
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 1, 2, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.desc()
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 2, 1 )
+		);
+	}
+
+	@Test
+	public void singleField_stringFieldBridge() throws Exception {
+		Query query = builder().all().createQuery();
+
+		// Missing value is not provided; the missing values should be considered as 0
+
+		Sort sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.asc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test(expected = SearchException.class)
+	public void singleField_stringFieldBridge_missingValue_use() throws Exception {
+		builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.onMissingValue().use( "1.5" )
+				.createSort();
+	}
+
+	@Test
+	public void singleField_numericFieldBridge() throws Exception {
+		Query query = builder().all().createQuery();
+
+		// Missing value is not provided; the missing values should be considered as 0
+
+		Sort sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+						.asc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test
+	public void singleField_numericFieldBridge_missingValue_use() throws Exception {
```

Comment:
I don't see myself committing such a crime unless forced to, but I don't care :+1: